lib/signet/oauth_2/client.rb

```diff
@@ -782,7 +782,13 @@ def expires_at
       # @param [String,Integer,Time] new_expires_at
       #    The access token issuance time.
       def expires_at=(new_expires_at)
-        @expires_at = normalize_timestamp(new_expires_at)
+        new_timestamp = normalize_timestamp new_expires_at
+        if new_timestamp.nil?
+          self.expires_in = nil
+        else
+          self.expires_in = new_timestamp - Time.now
```

Comment:
My read on the definition of `expires_in` is that it is the expiration time from `issued_at`, not from the current time. (See the `expires_at` method and how it computes its value from `expires_in`).

Reply:
It's possible we don't want to do it this way, but I was relying on the expires_in setter, which sets `issued_at` to `Time.now` anyway. It should also nil out `expires_at` in the process.

Reply:
Oh I see. That seems like a behavior change, though—previously, setting `expires_at` didn't reset `issued_at`, nor are we documenting anywhere that it will do so. It's not clear to me whether or not it _should_ reset the issue time, and barring a very good reason, I think we should err on the side of not changing the behavior.

Now that I look at this code in greater detail, the existence of separate `@expires_at` and `@expires_in` member variables themselves is going to be the source of a lot of confusion, since it's redundant. Can we just get rid of one altogether (while keeping the existing methods)?